static/main.js

```diff
@@ -13,32 +21,38 @@ function init() {
     input.addEventListener("input", reprocess, false);
     permalink.addEventListener("click", setSearch);
 
-    let match = (/(?:^\?|&)text=([^&]*)/).exec(window.location.search);
-    if (match) {
-        input.value = decodeURIComponent(match[1]);
+    let match;
+    if ((match = argument("text"))) {
+        input.value = decodeURIComponent(match[2]);
     }
 
     const macros = {};
-    // TODO: Add toggle for displayMode.
-    // https://github.com/Khan/KaTeX/issues/1035
     const options = {displayMode: true};
-    const macroRegex = /(?:^\?|&)(?:\\|%5[Cc])([A-Za-z]+)=([^&]*)/g;
+    const macroRegex = argumentRegex("(?:\\\\|%5[Cc])[A-Za-z]+|.");
     let macroString = "";
```

Comment:
The name `macroString` seems to be misleading.

Reply:
Maybe `optionsString` may be more appropriate.

Reply:
I've rewritten the code to avoid this recopying of the search string.